### PR TITLE
Upgrading to rails 4.2

### PR DESCRIPTION
**Pull request by jonnyjava**

### CODE REVIEW

File: config/secrets.yml [hunk 1]
```diff
@@ -10,11 +10,13 @@
 # Make sure the secrets in this file are kept private
 # if you're sharing your code publicly.
 
-production:
-  secret_key_base: <%= ENV["SECRET_KEY_BASE"] %>
-
 development:
-  secret_key_base: 526701f5b66b8163622e6358f6989f8ee10b27b41dcbdf587f4075208d3efe9b40da8ce5e82bf1e04937c1e4f89fcd3630f496aefcaa70138113c228e0af8d67
+  secret_key_base: ebc040546747a3f1ccdc345ba503efc5e6dff153319f34ca5cb30b500ee513bfaed4c6f89551f9da7eb2df728e83938eebf813071b0d03b687c0d5deb1a78db6
```
Comment (gponsu) [inline config/secrets.yml]: I think it's not a good practice to expose the tokens, even for development or test environments

File: spec/controllers/public_form_controller_spec.rb [hunk 2]
```diff
@@ -41,8 +41,8 @@
       it 'should re-render the form' do
```
Comment (gponsu) [inline spec/controllers/public_form_controller_spec.rb]: I think this test can be improved, perhaps with something like this:
```
expect { post :create, :garage => garage_params }.to change(Garage, :count).by(1) 
```

File: spec/controllers/tyre_fees_controller_spec.rb [hunk 3]
```diff
@@ -13,22 +13,22 @@
...
-      assigns(:tyre_fee).should be_a_new(TyreFee)
+      expect(assigns(:tyre_fee)).to be_a_new(TyreFee)
     end
   end
 
   describe 'GET edit' do
     it 'assigns the requested tyre_fee as @tyre_fee' do
       get :edit, { garage_id: garage, id: tyre_fee.to_param }
-      assigns(:tyre_fee).should eq(tyre_fee)
+      expect(assigns(:tyre_fee)).to eq(tyre_fee)
```
Comment (gponsu) [inline spec/controllers/tyre_fees_controller_spec.rb]: Using `assigns` in test controllers I think is deprecated: https://github.com/rails/rails/issues/18950

File: spec/spec_helper.rb [hunk 4]
```diff
@@ -50,5 +52,5 @@ def json(body)
 end
 
 def api_get path, *args
```
Comment (gponsu) [inline spec/spec_helper.rb]: I think this could go on `spec/support` folder.

File: spec/controllers/tyre_fees_controller_spec.rb (hunk 3 above)
Comment (jonnyjava) [inline spec/controllers/tyre_fees_controller_spec.rb]: Ok but.. how can I change it? I would test that a new instance variable is returned...

File: spec/controllers/public_form_controller_spec.rb (hunk 2 above)
Comment (jonnyjava) [inline spec/controllers/public_form_controller_spec.rb]: sadly public form will be dropped

File: spec/controllers/tyre_fees_controller_spec.rb (hunk 3 above)
Comment (gponsu) [inline spec/controllers/tyre_fees_controller_spec.rb]: DHH on the link:

> Testing what instance variables are in September by your controller is a bad idea. That's grossly overstepping the boundaries of what the test should know about. You can test what cookies are set, what HTTP code is returned, how the view looks, or what mutations Happened to the DB, but testing the innards of the controller is just not a good idea.

File: spec/controllers/public_form_controller_spec.rb (hunk 2 above)
Comment (gponsu) [inline spec/controllers/public_form_controller_spec.rb]: This comment was for the test of the line 25.

**Pull request merged**